app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,86 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import androidx.annotation.StringRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.ContentAlpha
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    val isSelected: Boolean,
+    val categoryText: MovieCategoryLabelTextViewState
+)
+
+sealed class MovieCategoryLabelTextViewState{
+    class InputText(val text: String):MovieCategoryLabelTextViewState()
+    class ResourceText(@StringRes val textRes: Int): MovieCategoryLabelTextViewState()
+}
+
+@Composable
+fun MovieCategoryLabel(
+    movieCategoryLabelViewState: MovieCategoryLabelViewState,
+    onClick: () -> Unit
+){
+    if(movieCategoryLabelViewState.isSelected){
+        Column(
+            modifier = Modifier.width(intrinsicSize = IntrinsicSize.Max)
+                               .clickable{onClick()}
+        ){
```

Comment:
Moved the Coulmn to MovieCategoryLabel function. Also added lambda as function parameter.